nebari-dev/nebari
# template the kustomization.yaml file

## Reference Issues or PRs
part of #2557 


## What does this implement/fix?
Adds the ability to template the `kustomization.yaml` file to allow variables from the config to be dynamically set.

_Put a `x` in the boxes that apply_


## Testing


## How to test this PR?
1. enable healthchecks with
```yaml
monitoring:
  healthchecks:
    enabled: true:
```
2. run a render with `nebari render -c nebari-config.yaml`
3. check the rendered objects namespace with `grep -r  namespace stages/10-kubernetes-kuberhealthy/` and `grep -r  namespace stages/11-kubernetes-kuberhealthy-healthchecks/` and verify that the correct namespace has been set (it should be `dev`) by default.
4. change the namespace to a different namespace in your `nebari-config.yaml` file and rerun the render
5. using the above commands, verify the namespace was updated.
6. return to your original namespace and rerun the render
7. run a `nebari deploy` to validate that everything deploys to the correct namespace


## Any other comments?
